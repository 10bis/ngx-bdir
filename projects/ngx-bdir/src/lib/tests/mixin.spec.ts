const sass = require('node-sass');

describe('mixins test', () => {
  function setup(source) {
    return sass
      .renderSync({
        data: `
        @import 'projects/ngx-bdir/src/lib/styles/bdir.mixins.scss';
        ${source}
      `
      })
      .css.toString();
  }

  it('should create an rtl ltr support for margin end', () => {
    const source = `
      h1 {
        @include margin-end(20px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for padding start', () => {
    const source = `
      h1 {
        @include padding-start(20px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for padding end without encapsulation', () => {
    const source = `
      h1 {
        @include padding-end(20px, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for float', () => {
    const source = `
      div {
        @include float();
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for dir without encapsulation', () => {
    const source = `
      div {
        @include dir(true, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for start (left / right) positioning', () => {
    const source = `
      div {
        @include start(40px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for end (right / left) positioning without encapsulation', () => {
    const source = `
      div {
        @include end(20%, false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for translating ', () => {
    const source = `
      div {
        @include transformTranslate(5px, 5px);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });

  it('should create an rtl ltr support for mirroring without encapsulation', () => {
    const source = `
      div {
        @include mirror(false);
      }
    `;
    const css = setup(source);
    expect(css).toMatchSnapshot();
  });
});
