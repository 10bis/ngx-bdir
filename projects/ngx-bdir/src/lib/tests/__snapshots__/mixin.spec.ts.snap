// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixins test should create an rtl ltr support for dir without encapsulation 1`] = `
"*[dir=ltr] div {
  dir: left; }

*[dir=rtl] div {
  dir: right; }
"
`;

exports[`mixins test should create an rtl ltr support for end (right / left) positioning without encapsulation 1`] = `
"*[dir=ltr] div {
  left: 20%; }

*[dir=rtl] div {
  right: 20%; }
"
`;

exports[`mixins test should create an rtl ltr support for float 1`] = `
":host-context(*[dir=ltr]) div {
  float: left; }

:host-context(*[dir=rtl]) div {
  float: right; }
"
`;

exports[`mixins test should create an rtl ltr support for margin end 1`] = `
":host-context(*[dir=ltr]) h1 {
  margin-right: 20px; }

:host-context(*[dir=rtl]) h1 {
  margin-left: 20px; }
"
`;

exports[`mixins test should create an rtl ltr support for mirroring without encapsulation 1`] = `
"*[dir=ltr] div {
  transform: scale(1,1); }

*[dir=rtl] div {
  transform: scale(-1,1); }
"
`;

exports[`mixins test should create an rtl ltr support for padding end without encapsulation 1`] = `
"*[dir=ltr] h1 {
  padding-right: 20px; }

*[dir=rtl] h1 {
  padding-left: 20px; }
"
`;

exports[`mixins test should create an rtl ltr support for padding start 1`] = `
":host-context(*[dir=ltr]) h1 {
  padding-left: 20px; }

:host-context(*[dir=rtl]) h1 {
  padding-right: 20px; }
"
`;

exports[`mixins test should create an rtl ltr support for start (left / right) positioning 1`] = `
":host-context(*[dir=ltr]) div {
  left: 40px; }

:host-context(*[dir=rtl]) div {
  right: 40px; }
"
`;

exports[`mixins test should create an rtl ltr support for translating  1`] = `
":host-context(*[dir=ltr]) div {
  transform: translate(5px,5px); }

:host-context(*[dir=rtl]) div {
  transform: translate(-5px,5px); }
"
`;
